### PR TITLE
Enable JCK on Windows 

### PR DESCRIPTION
**Pull request by Mesbah-Alam**
Resolves https://github.com/AdoptOpenJDK/openjdk-systemtest/issues/279 
Signed-off-by: Mesbah_Alam@ca.ibm.com <Mesbah_Alam@ca.ibm.com>

### CODE REVIEW

**Comment by Mesbah-Alam** [comment]
Update Summary: 

1) Adds JCK native compilation steps in openjdk.test.jck/build.xml 
2) Updates JCK makefile for Cygwin 
3) Updates Jck.java as we have changed the directory name in which jck materials get checked out fromn `jck<versionNo>` to `jck<versionNo>-unzipped`.

**Comment by Mesbah-Alam** [comment]
Tests , Linux : https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Grinder_Advanced/204/console (passes)

Windows: https://hyc-runtimes-jenkins.swg-devops.com/view/Test_grinder/job/Grinder_Advanced/206/
(fails with unrelated issue?)

File: openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java [hunk 1]
```diff
@@ -145,16 +146,18 @@ public void pluginInit(StfCoreExtension test) throws Exception {
 		StfTestArguments testArgs = test.env().getTestProperties("tests","jckversion","testsuite","config=[NULL]","executiontype=[multijvm]","withagent=[off]","interactive=[no]","jckRoot=[NULL]","concurrency=[NULL]");
 		
 		testJdk = System.getenv("JAVA_HOME");
-		tests = testArgs.get("tests");
+		tests = testArgs.get("tests");	
+		jckVersion = testArgs.get("jckversion");	
+		jckVersionNo = jckVersion.replace("jck", "");		
+		jckUnzippedFolderName = "JCK" + jckVersionNo + "-unzipped";
```
Comment (llxia) [inline openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: Same here. The folder name should be passed in.

File: openjdk.test.jck/build.xml [hunk 2]
```diff
@@ -88,6 +103,32 @@ limitations under the License.
 			<include name="**/*.java"/>
 		</javac>
 	</target>
+	
+	<target name="build-natives" depends="check-prereqs, setup-native-build-command, build-natives-windows, build-natives-unix">
+	</target>
+	
+	<target name="setup-native-build-command">
+		<echo message="building natives for java_platform ${java_platform}"/>
+		<property name="openjdk_test_jck_native_build_command" value='${setup_windows_build_env}make build -f ${jck_makefile_dir}/makefile SRCDIR=${env.JCK_ROOT}/JCK${jck_short_version}-unzipped/JCK-runtime-${jck_short_version} PLATFORM=${platform_short_name} JAVA_HOME=${TEST_JDK_HOME} OUTDIR=${env.JCK_ROOT}/JCK${jck_short_version}-unzipped/natives'/>
```
[the remark this answers is not in the document]
Comment (Mesbah-Alam) [reply openjdk.test.jck/build.xml]: Files updated! Thanks.

File: openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java (hunk 1 above)
Comment (Mesbah-Alam) [reply openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: I believe this is different than the SRCDIR and OUTDIR in the build files. This JCK main folder name is being constructed only using JCK version, which is being passed via the JCK Playlist command linens where Jck.java is invoked. Whoever is running this class should have JCK staged ahead of time, which means the JCK-<jckVersionnNo-unzipped folder would be available. So, I am not sure if we really have to update all the playlists and pass in the unzipped folder name.

Comment (llxia) [reply openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: Talked with Mesbah, the main idea is to allow users to use the script if they have a different jck repo/dir.

Comment (Mesbah-Alam) [reply openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: Hi Lan, changes have been made to the PR. Please check. (sent grinder links to your slack).

File: openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java [hunk 3]
```diff
@@ -181,24 +184,20 @@ public void pluginInit(StfCoreExtension test) throws Exception {
...
-		nativesLoc = test.env().findPrereqDirectory(jckTopDir + jckVersion + "/natives/" + test.env().getPlatform());
+		jckBase = test.env().findPrereqDirectory(testSuiteFolder);
+		nativesLoc = test.env().findPrereqDirectory("natives/" + test.env().getPlatform());
 
 		DirectoryRef repositoryConfigLoc = test.env().findTestDirectory("openjdk.test.jck/config");
 		jtiFile = repositoryConfigLoc.childFile("/" + jckVersion + "/" + testSuite.toString().toLowerCase() + ".jti");
+		//fileUrl = "file:///" + jckRoot + "/" + testSuiteFolder + "/testsuite.jtt";
 		
-		fileUrl = "file:///" + test.env().findPrereqFile(jckTopDir + jckVersion + "/" + testSuiteFolder + "/testsuite.jtt");
+		fileUrl = "file:///" + test.env().findPrereqFile(testSuiteFolder + "/testsuite.jtt");
```
Comment (llxia) [inline openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: Should we use `jckRoot` here?

File: openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java [hunk 4]
```diff
@@ -339,10 +338,10 @@ public void execute(StfCoreExtension test) throws Exception {
 					addModules = "--add-modules java.xml.ws,java.corba";
 				}
 				javatestAgent = test.doRunBackgroundProcess("Starting javatest agent", "AGNT", ECHO_ON, ExpectedOutcome.neverCompletes(), test.createJavaProcessDefinition()
-						.addJvmOption("-Djavatest.security.allowPropertiesAccess=true -Djava.security.policy=" + test.env().findPrereqFile(jckTopDir + jckVersion + "/" + testSuiteFolder + "/lib/jck.policy").toString())
+						.addJvmOption("-Djavatest.security.allowPropertiesAccess=true -Djava.security.policy=" + test.env().findPrereqFile(testSuiteFolder + "/lib/jck.policy").toString())
```
Comment (llxia) [inline openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: Should we include `jckRoot`? Without it, is it possible to find the file in the wrong jck version? Same for the rest of `findPrereqFile()`

File: openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java (hunk 3 above)
Comment (Mesbah-Alam) [reply openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: When JckRoot is provided as argument to Jck.java, which is the case for us here, jckRoot == jckTopDir.

File: openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java (hunk 4 above)
Comment (Mesbah-Alam) [reply openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: When JckRoot is provided as argument to Jck.java, which is the case for us here, jckRoot == jckTopDir.

Comment (llxia) [reply openjdk.test.jck/src/test.jck/net/adoptopenjdk/stf/Jck.java]: Thanks for clarifying, Mesbah. :)

**Pull request merged**